torchgeo/datasets/eudem.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""European Digital Elevation Model (EU-DEM) dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, Optional
+
+import matplotlib.pyplot as plt
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import check_integrity, extract_archive
+
+
+class EUDEM(RasterDataset):
+    """European Digital Elevation Model (EU-DEM) Dataset.
+
+    The `EU-DEM
+    <https://land.copernicus.eu/imagery-in-situ/eu-dem/eu-dem-v1.1?tab=mapview>`_
+    dataset is a Digital Elevation Model of reference for the entire European region.
+    The dataset can be downloaded from this `website
+    <https://land.copernicus.eu/imagery-in-situ/eu-dem/eu-dem-v1.1?tab=mapview>`_
+    after making an account. A dataset factsheet is available
+    `here <https://land.copernicus.eu/user-corner/publications/eu-dem-flyer/view>`.
+
+    Dataset features:
+
+    * DEMs at 25 m per pixel spatial resolution (~40,000x40,0000 px)
+    * vertical accuracy of +/- 7 m RMSE
+    * data fused from `ASTER GDEM
+      <https://lpdaac.usgs.gov/news/nasa-and-meti-release-aster-global-dem-version-3/>`_,
+      `SRTM <https://www2.jpl.nasa.gov/srtm/>`_ and Russian topomaps
+
+    Dataset format:
+
+    * DEMs are single-channel tif files
+
+    If you use this dataset in your research, please give credit to:
+
+    * `Copernicus <https://land.copernicus.eu/imagery-in-situ/eu-dem/eu-dem-v1.1>`_
+
+    .. versionadded:: 0.3
+    """
+
+    is_image = False
+    filename_glob = "eu_dem_v11_*.TIF"
+    zipfile_glob = "eu_dem_v11_*[A-Z0-9].zip"
+    filename_regex = r"""(?P<name>[eudem_v11]{10})_(?P<id>[A-Z0-9]{6})"""
```

Comment:
If this is on a single line, there's no need to use triple quotes. Also, I don't think it needs to be a raw string since it doesn't contain any backslashes.